RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitSoAFromCUDA.cc

```diff
@@ -69,29 +73,37 @@ template <typename TrackerTraits>
 void SiPixelRecHitSoAFromCUDAT<TrackerTraits>::acquire(edm::Event const& iEvent,
                                                        edm::EventSetup const& iSetup,
                                                        edm::WaitingTaskWithArenaHolder waitingTaskHolder) {
-  cms::cuda::Product<TrackingRecHit2DGPUT<TrackerTraits>> const& inputDataWrapped = iEvent.get(hitsTokenGPU_);
+  cms::cuda::Product<HitsOnDevice> const& inputDataWrapped = iEvent.get(hitsTokenGPU_);
   cms::cuda::ScopedContextAcquire ctx{inputDataWrapped, std::move(waitingTaskHolder)};
   auto const& inputData = ctx.get(inputDataWrapped);
 
   nHits_ = inputData.nHits();
+  hits_h_ = HitsOnHost(nHits_, ctx.stream());
+  cudaCheck(cudaMemcpyAsync(hits_h_.buffer().get(),
+                            inputData.const_buffer().get(),
+                            inputData.bufferSize(),
+                            cudaMemcpyDeviceToHost,
+                            ctx.stream()));  // Copy data from Device to Host
   LogDebug("SiPixelRecHitSoAFromCUDA") << "copying to cpu SoA" << inputData.nHits() << " Hits";
 
   if (0 == nHits_)
     return;
```

Comment:
This check was used to protect the following copies.
Once they are removed the check can be removed as well.